pygmt/datasets/samples.py

```diff
@@ -95,51 +95,31 @@ def load_sample_data(name):
     return data
 
 
-def load_japan_quakes(**kwargs):
+def _load_japan_quakes():
     """
-    (Deprecated) Load a table of earthquakes around Japan as a
-    pandas.DataFrame.
-
-    .. warning:: Deprecated since v0.6.0. This function has been replaced with
-       ``load_sample_data(name="japan_quakes")`` and will be removed in
-       v0.9.0.
-
-    Data is from the NOAA NGDC database. This is the ``@tut_quakes.ngdc``
-    dataset used in the GMT tutorials.
-
-    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
-    first time you invoke this function. Afterwards, it will load the data from
-    the cache. So you'll need an internet connection the first time around.
+    Load a table of earthquakes around Japan as a pandas.DataFrame.
```

Comment:
```suggestion
    Load a table of earthquakes around Japan as a pandas.DataFrame.
    
    Data is from the NOAA NGDC database. This is the "@tut_quakes.ngdc"
    dataset used in the GMT tutorials.
```

Reply:
@yvonnefroehlich Thanks for your comments. 

> Data is from the NOAA NGDC database. 

I've added this sentence back in 92bcec34cce00980b89f574aaec343c8167c56c6.


> This is the "@tut_quakes.ngdc" dataset used in the GMT tutorials.

I don't think this sentence would provide any meaningful information. The file name is easy to get from the function and we all know that these dataset are used in the tutorials.

Reply:
I was not complete sure about this, as in the PRs #2306 and #2308 this is not removed but in the PRs #2301 (this PR), #2303, and #2305 it is removed. I have big preference on this, but it should be consistent for all functions.

Reply:
Let's discuss it in this PR and then update other PRs if necessary.

@michaelgrund and @willschlitzer what do you think?

Reply:
I'm with @seisman since this information is not really required at this point in my opinion.

Reply:
OK, then we can merge this PR.